pandas/tests/util/test_assert_produces_warning.py

```diff
@@ -95,18 +95,21 @@ def test_catch_warning_category_and_match(category, message, match):
 
 
 @pytest.mark.parametrize(
-    "message, match",
+    "match",
     [
-        ("Warning message", "Not this message"),
-        ("Warning message", "warning"),
-        ("Warning message", r"\d+"),
+        ("Not this message"),
+        ("Warning"),
+        (r"\d+"),
     ],
 )
-def test_fail_to_match(category, message, match):
-    msg = f"Did not see warning {repr(category.__name__)} matching"
-    with pytest.raises(AssertionError, match=msg):
+def test_fail_to_match(category, match):
+    msg1 = "This is not a match."
+    msg2 = "Another unmatched warning."
+    unmatched = rf"{category.__name__}\('{msg1}'\), {category.__name__}\('{msg2}'\)"
```

Comment:
can you make three distinct tests here, and match the full `unmatched` error message each time?

Reply:
Thank you for the feedback! Yes, I can do that. I think I see the reasoning for the original three test cases that I could preserve.